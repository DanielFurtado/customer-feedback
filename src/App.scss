* {
  font-family: sans-serif;
}

.App--header {
  align-items: center;
  background-color: rgb(12, 17, 66);
  display: flex;
  height: 100px;
  padding: 20px;
  width: 100%;
}

.Main--container {
  align-items: center;
  display: flex;
  flex-flow: column;
  justify-content: center;
  margin: 0 auto;
  max-width: 500px;
  padding: 20px;
  width: 100%;
}

.Button--primary {
  cursor: pointer;
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
  text-align: center;
  text-decoration: none;
  color: rgb(12, 17, 66);
  margin: 20px 0;
  padding: 12px 24px;
  background: rgb(43, 212, 219);
  // background: #22a77d;
  border: none;
  border-radius: 8px;

  &:hover {
    background: rgb(107, 225, 230);
  }
}

