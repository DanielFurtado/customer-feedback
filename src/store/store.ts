import { configureStore } from '@reduxjs/toolkit';
import ratingReducer from './rating/rating.reducer';

export const store = configureStore({
  reducer: {
    feedbackList: ratingReducer
  }
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
