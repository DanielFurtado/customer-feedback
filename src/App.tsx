import './App.css';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        Checkout.com
      </header>
    </div>
  );
}

export default App;
