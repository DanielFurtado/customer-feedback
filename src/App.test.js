import {BrowserRouter as Router} from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from './store/store';
import renderer from 'react-test-renderer';
import App from './App';

it('renders', () => {
  const component = renderer.create(
    <Provider store={store}>
      <Router>
        <App />
      </Router>
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
