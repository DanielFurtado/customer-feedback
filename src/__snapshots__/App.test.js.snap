// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="sc-hLBbgP ccLVbk"
>
  <header
    className="sc-bcXHqe gAzLXc"
    role="banner"
  >
    <nav
      className="sc-gswNZR cYZNu"
    >
      <a
        aria-current="page"
        className="Nav--logo-container active"
        href="/"
        onClick={[Function]}
      >
        <svg>
          logo.svg
        </svg>
      </a>
      <ul
        className="sc-dkrFOg cybPPa"
        role="menubar"
      >
        <li
          role="none"
        >
          <a
            aria-current="page"
            className="Nav--link active"
            href="/"
            onClick={[Function]}
            role="menuitem"
          >
            Form
          </a>
        </li>
        <li
          role="none"
        >
          <a
            className="Nav--link"
            href="/results"
            onClick={[Function]}
            role="menuitem"
          >
            Results
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main
    className="sc-eDvSVe ghWFaa"
  >
    <form
      className="sc-gKPRtg fKCcpJ"
      noValidate={true}
    >
      <h2>
        Feedback Form
      </h2>
      <div
        className="sc-iBYQkv LVoZX"
      >
        <div
          className="sc-ftTHYK fjpkYH"
        >
          <div
            className="Form--group"
          >
            <label
              htmlFor="name"
            >
              Name *
            </label>
            <input
              aria-invalid="false"
              aria-label="Name"
              id="name"
              name="name"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Name"
              type="text"
            />
          </div>
          <div
            className="Form--group"
          >
            <label
              htmlFor="email"
            >
              Email *
            </label>
            <input
              aria-invalid="false"
              aria-label="Email"
              id="email"
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Email"
              type="text"
            />
          </div>
          <div
            className="Form--group"
          >
            <label
              htmlFor="rating"
            >
              Rating *
            </label>
            <input
              aria-invalid="false"
              aria-label="Rating"
              id="rating"
              max="5"
              min="1"
              name="rating"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Rating (1-5 stars)"
              type="number"
            />
          </div>
        </div>
        <div
          className="sc-ftTHYK fjpkYH"
        >
          <div
            className="sc-pyfCe eOVupX"
          >
            <label
              htmlFor="comment"
            >
              Comment *
            </label>
            <textarea
              aria-invalid="false"
              aria-label="Comment"
              id="comment"
              name="comment"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Comment"
            />
          </div>
        </div>
      </div>
      <div
        className="sc-jrcTuL deiqno"
      >
        <button
          className="sc-jSUZER eFvONg"
          onClick={[Function]}
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </main>
</div>
`;
